crates/sncast/src/main.rs

```diff
@@ -740,3 +735,23 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
         clone_or_else!(cli.wait_timeout, config.wait_params.get_timeout()),
     );
 }
+
+fn get_cast_config(cli: &Cli) -> Result<CastConfig> {
+    let global_config_path = get_global_config_path().unwrap_or_else(|err| {
+        eprintln!("Error getting global config path: {err}");
+        Utf8PathBuf::new()
+    });
+
+    let global_config =
+        load_global_config::<CastConfig>(&Some(global_config_path.clone()), &cli.profile)
+            .unwrap_or_else(|_| {
+                load_global_config::<CastConfig>(&Some(global_config_path), &None).unwrap()
+            });
+
+    let local_config = load_global_config::<CastConfig>(&None, &cli.profile)?;
+
+    let mut combined_config = combine_cast_configs(&global_config, &local_config);
+
+    update_config_with_cli(&mut combined_config, cli);
```

Comment:
Can we change the function `update_config_with_cli` to something like `config_with_cli` or similar and make it return a new config instead of taking a mutable reference?